codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -108,6 +108,20 @@ module Make = (
     }
   }
 
+  let getLogSelection = (~contractAddressMapping): array<LogSelection.t> => {
+    T.contracts->Belt.Array.keepMap((contract): option<LogSelection.t> => {
+      switch contractAddressMapping->ContractAddressingMap.getAddressesFromContractName(
+        ~contractName=contract.name,
+      ) {
+      | [] => None
+      | addresses =>
+        let topicSelection = LogSelection.makeTopicSelection(~topic0=contract.sighashes)->Utils.unwrapResultExn
+
```

Comment:
Can we rather not unwrap here and then we don't need a try catch block below?

Reply:
Then we'll need to convert `array<result<_>>` to `result<array<>>`, so I prefer throwing exception. The array is not big here, but still I don't want to add unnecessary performance overhead.